src/components/Buttons/Button.js

```diff
@@ -0,0 +1,266 @@
+        }
+      }
+    },
+
+    disabled: {
+      base: {
+        backgroundColor: colors.GRAY_74,
+        color: colors.WHITE,
+        cursor: 'not-allowed'
+      }
+    }
+  }
+}
+
+const Button = props => {
+  // Disabled styles override all other snacks-specific button styles
+  const snacksStyle = props.disabled ? 'disabled' : props.snacksStyle
+  const snacksStyles = getSnacksStyles(props)
+
+  const ElementType = props.href ? 'a' : 'button'
+
+  const finalProps = {
+    disabled: props.disabled,
+    tabIndex: props.tabIndex,
+    type: props.type,
+    style: [
+      baseStyles,
+      sizeStyles[props.size],
+      snacksStyles[snacksStyle].base,
+      props.inverted && snacksStyles[snacksStyle].inverted,
+      ElementType === 'a' && linkStyles,
+      props.styles
+    ],
+    onClick: e => {
+      e.preventDefault()
+      props.onClick()
+    },
+    ...props.extraProps
+  }
+  if (props.href) {
+    finalProps.href = props.href
+  }
+
+  let icon = props.icon
+  if (typeof props.icon === 'string') {
+    icon = <Icon name={props.icon} />
+  }
+
+  if (icon && props.iconPosition === 'left') {
+    return (
+      <ElementType {...finalProps}>
+        {icon}
+        <span style={{ ...spacing.MARGIN_RIGHT_XS }} />
+        {props.children}
+      </ElementType>
+    )
+  }
+
+  if (icon && props.iconPosition === 'right') {
+    return (
+      <ElementType {...finalProps}>
+        {props.children}
+        <span style={{ ...spacing.MARGIN_LEFT_XS }} />
+        {icon}
+      </ElementType>
+    )
+  }
+
+  return <ElementType {...finalProps}>{props.children}</ElementType>
+}
+
+Button.propTypes = {
+  /** Size of the button. */
+  size: PropTypes.oneOf(['tiny', 'small', 'standard', 'large']),
+
+  /** Snacks button variants. */
+  snacksStyle: PropTypes.oneOf(['primary', 'secondary', 'coupon']),
+
+  /** Optional style overrides. */
+  styles: PropTypes.object,
+
+  /** Whether or not the button is disabled. */
+  disabled: PropTypes.bool,
+
+  /** Sets the HTML type attribute on the element. */
+  type: PropTypes.oneOf(['button', 'reset', 'submit']),
+
+  /**
+    An optional icon. Can be a an icon name
+    or a Snacks `Icon` component.
+  */
+  icon: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+
+  /** Where the icon is positioned relative to primary text content. */
+  iconPosition: PropTypes.oneOf(['left', 'right']),
+
+  /** Callback that fires when the button is clicked. */
+  onClick: PropTypes.func,
+
+  /** Reverses colors. Use for rendering buttons on dark backgrounds. */
+  inverted: PropTypes.bool,
+
+  /** Sets the HTML `tabindex` attribute. */
+  tabIndex: PropTypes.number,
+
+  /**
+    If href is provided, `Button` will render as an anchor
+    tag with the supplied href value.
+  */
+  href: PropTypes.string,
+
+  /** Text content for the button. */
+  children: PropTypes.node,
+
+  /** Any additonal props to add to the element (e.g. data attributes). */
+  extraProps: PropTypes.object,
```

Comment:
I think this name is a bit confusing. They're almost explicitly not props, so I think calling something like `elementAttributes` or something would be more clear

Reply:
Funny enough, i actually originally called it `elementAttributes` and then changed it :D